src/store/reducers/room.ts

```diff
@@ -17,12 +20,86 @@ function userReducer(state: ConnectedRoomMap = initialState, action: RoomAction)
             return roomsToKeep;
         case UPDATE_ROOM:
             const roomIdToUpdate = action.payload;
-            return Object.assign({}, state, {
```

Comment:
Uhh, just realized this while coding but I had this because if you do it the way you edited, the page won't re-render because it doesn't realize that the internals of the objects has changed. Maybe the way I did it is not the best but we shouldn't leave it like this.